src/pages/guides/tutorials/stats-addon.md

```diff
@@ -0,0 +1,1108 @@
+    }
+  },
+  { // page 2
+    "dimensions": { "width": 800, "height": 400 },
+    "nodes": {
+      "ab:Artboard": 1, "MediaContainer": 4, "Rectangle": 2
+    }
+  },
+  // ...
+]
+```
+
+The various `"ab:Artboard"`, `"MediaContainer"` and others, are the Node type strings as Adobe Express exposes them. Let's create the `getDocumentData()` function that outputs such a structure.
+
+![](images/stats-addon-type.png)
+
+<CodeBlock slots="heading, code" repeat="1" languages="code.js"/>
+
+#### documentSandbox/code.js
+
+```js
+import { getNodeData } from "./utils";
+
+runtime.exposeApi({
+  async getDocumentData() {
+    const doc = editor.documentRoot;      // get the document
+    let documentData = [];                // initialize the array to return
+    for (const page of doc.pages) {       // loop through each page
+      let pageData = {};                  // create an empty object
+      pageData.dimensions = {             // get and store the page `dimensions`
+        width: page.width,
+        height: page.height,
+      };
+      pageData.nodes = getNodeData(page); // 👈 build the `nodes` object (more on this later)
+      documentData.push(pageData);        // push the object to the documentData array
+    }
+    // invoke the iframe method to create the table on the UI
+    await panelUIProxy.createTable(documentData);
+  },
+});
+```
+
+The code comments will guide you through the process of getting the document, loop through pages extracting dimensions, and retrieving nodes metadata, but up to a point. What's `getNodeData`? As I mentioned before, I've split the Document API code into two parts: the main Document Sandbox entrypoint (`code.js`, where methods are exposed to and imported from the iframe) and `table-utils.js`, which is kept private to the context and exports just what `code.js` needs—the `getNodeData()` method, which makes use of `increaseCount()`.
+
+<CodeBlock slots="heading, code" repeat="1" languages="utils.js"/>
+
+#### documentSandbox/table-utils.js
+
+```js
+const increaseCount = (obj, type) => {
+  // If the type is already in the object, increase its count, otherwise set it to 1
+  if (obj.hasOwnProperty(type)) {
+    obj[type] += 1;
+  } else {
+    obj[type] = 1;
+  }
+};
+
+const getNodeData = (node, nodeData = {}) => {
+  if (node.type === "MediaContainer") {
+    return nodeData;
+  }
+
+  // Check if the current node has children and if they are not an empty array
+  if (node.allChildren && node.allChildren.length > 0) {
+    // Iterate over all children using for..of
+    for (const child of node.allChildren) {
+      // Increase the count for the current type
+      increaseCount(nodeData, child.type);
+
+      // Recursively call getNodeData for each child that has its own children
+      // ... unless it's a MediaContainer
+      if (
+        child.type !== "MediaContainer" &&
+        child.allChildren &&
+        child.allChildren.length > 0
```

Comment:
Might be good to remind people that `allChildren` returns an interator, not an array. So if they want to use array methods, they need to use `Array.from(allChildren)`.

Reply:
Sure! When I wrote this, I believe it wasn't an iterator yet :-)